### PR TITLE
feat/create-customer-use-case

### PR DESCRIPTION
**Pull request by lucaschf**
## Summary by Sourcery

Implement a new customer creation use case with associated DTOs and error handling. Refactor error class names for consistency and add comprehensive tests for the new functionality.

New Features:
- Introduce a new use case for creating customers, including the necessary data transfer objects (DTOs) for customer creation and response.

Enhancements:
- Refactor error classes by renaming 'DuplicatedKeyRepositoryError' to 'DuplicateKeyError' and 'RecordNotFoundInRepositoryError' to 'RecordNotFoundError' for consistency.

Tests:
- Add tests for the new customer creation use case, covering scenarios such as successful creation, duplicate customer error, missing name, and missing required data.

### CODE REVIEW

File: tests/application/use_cases/customer/create/test_customer_create_use_case.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+async def test_create_customer_success() -> None:
+    customer_repository_mock = Mock(spec=ICustomerRepository)
+    create_customer_use_case_fx = CreateCustomerUseCase(customer_repository_mock)
+
+    customer_repository_mock.find.return_value = None
+    customer_data = CustomerCreationDTO(
+        name="John Doe",
+        email=EmailAddress(address="some@example.com"),
+        cpf=CPF(number="10856446696"),
+    )
```
[the remark this answers is not in the document]
Comment (lucaschf) [reply tests/application/use_cases/customer/create/test_customer_create_use_case.py]: removed find use
